src/cljc/rems/common/catalogue_util.cljc

```diff
@@ -18,6 +18,16 @@
   (when (and enable-ega (ega-dataset? resid))
     (str (or ega-organization "https://ega-archive.org/datasets/") resid)))
 
+;; https://www.crossref.org/blog/dois-and-matching-regular-expressions/
+;; "For the 74.9M DOIs we have seen this matches 74.4M of them. If you need to use only one pattern then use this one."
+(defn- doi? [resid]
+  (and resid (re-find #"(?i)10.\d{4,9}[-._;()/:A-Z0-9]+$" resid)))
```

Comment:
the regex is slightly modified from the one mentioned in the URL. i removed start of string character to allow matching URLs, in case URL is provided instead of resource id. this allows matching an DOI and extracting only the id, which is then later reconstructed into an URL